integration_tests/src/testcases/locator.test.ts

```diff
@@ -0,0 +1,455 @@
+  });
+  it('Query factory instances', async () => {
+    const { locator } = context.contracts;
+    const res = await locator.queryFactoryInstances();
+    expect(res).toHaveLength(2);
+    for (const instance of res) {
+      expect(instance.addr).toHaveLength(66);
+      expect(instance.contracts.token_contract).toHaveLength(66);
+      expect(instance.contracts.core_contract).toHaveLength(66);
+      expect(instance.contracts.puppeteer_contract).toHaveLength(66);
+      expect(instance.contracts.staker_contract).toHaveLength(66);
+      expect(instance.contracts.withdrawal_voucher_contract).toHaveLength(66);
+      expect(instance.contracts.withdrawal_manager_contract).toHaveLength(66);
+      expect(instance.contracts.strategy_contract).toHaveLength(66);
+      expect(instance.contracts.validators_set_contract).toHaveLength(66);
+      expect(instance.contracts.distribution_contract).toHaveLength(66);
+      expect(instance.contracts.rewards_manager_contract).toHaveLength(66);
+    }
+    let matches: number = 0;
+    for (const factory of context.contracts.factories) {
+      const factoryState = await factory.queryState();
+      for (const instance of res) {
+        if (instance.addr === factory.contractAddress) {
+          matches += 1;
+          expect(instance.contracts.token_contract).toBe(
+            factoryState.token_contract,
+          );
+          expect(instance.contracts.core_contract).toBe(
+            factoryState.core_contract,
+          );
+          expect(instance.contracts.puppeteer_contract).toBe(
+            factoryState.puppeteer_contract,
+          );
+          expect(instance.contracts.staker_contract).toBe(
+            factoryState.staker_contract,
+          );
+          expect(instance.contracts.withdrawal_voucher_contract).toBe(
+            factoryState.withdrawal_voucher_contract,
+          );
+          expect(instance.contracts.withdrawal_manager_contract).toBe(
+            factoryState.withdrawal_manager_contract,
+          );
+          expect(instance.contracts.strategy_contract).toBe(
+            factoryState.strategy_contract,
+          );
+          expect(instance.contracts.validators_set_contract).toBe(
+            factoryState.validators_set_contract,
+          );
+          expect(instance.contracts.distribution_contract).toBe(
+            factoryState.distribution_contract,
+          );
+          expect(instance.contracts.rewards_manager_contract).toBe(
+            factoryState.rewards_manager_contract,
+          );
+        }
+      }
+    }
+    expect(matches).toBe(res.length);
+  });
+  it('Try to remove factory instances from locator contract', async () => {
+    const { locator } = context.contracts;
+    const { account } = context;
+    await locator.removeChains(
+      account.address,
+      {
+        names: ['instance_2'],
+      },
+      'auto',
+      '',
+      [],
+    );
+    expect((await locator.queryChains()).length).toEqual(1);
```

Comment:
I'd recommend you to check name of the instance left in the locator to be fully sure.